source/common/network/connection_impl.cc

```diff
@@ -39,6 +40,90 @@ void ConnectionImplUtility::updateBufferStats(uint64_t delta, uint64_t new_total
   }
 }
 
+Api::SysCallIntResult DynamicSocketOptionsImpl::setSocketRecvBufferSize(int buffer_size) const {
+  int rc = -1;
+  int errsrv = -1;
+  socklen_t option_len = sizeof(int);
+  if (-1 == io_handle_.fd()) {
+    ENVOY_CONN_LOG(trace, "Stale socket file handle, operation Set Receive Buffer aborted", *this);
+    return {rc, errsrv};
+  }
+  // Modify the receive buffer size to 'buffer_size'
+  auto& os_syscalls = Api::OsSysCallsSingleton::get();
+  Api::SysCallIntResult result =
+      os_syscalls.setsockopt(io_handle_.fd(), SOL_SOCKET, SO_RCVBUF, &buffer_size, option_len);
+  if (-1 == result.rc_) {
+    ENVOY_CONN_LOG(trace, "Failed to modify socket receive buffer to size: buffer_size={}", *this,
+                   buffer_size);
+  }
+  return result;
+}
+
+Api::SysCallIntResult DynamicSocketOptionsImpl::getSocketRecvBufferSize(int& recv_buf_size) const {
+  int rc = -1;
+  int errsrv = -1;
+  socklen_t option_len = sizeof recv_buf_size;
+
+  if (-1 == io_handle_.fd()) {
+    ENVOY_CONN_LOG(trace, "Stale socket file handle, operation Get Receive Buffer Size aborted",
+                   *this);
+    return {rc, errsrv};
+  }
+
+  // Read the set receive buffer size as it is double the value configured through
+  // 'setSocketRecvBufferSize' on some machines this value can not be lower than '2304' bytes
+  auto& os_syscalls = Api::OsSysCallsSingleton::get();
+  Api::SysCallIntResult result =
+      os_syscalls.getsockopt(io_handle_.fd(), SOL_SOCKET, SO_RCVBUF, &recv_buf_size, &option_len);
+  if (-1 == result.rc_) {
+    ENVOY_CONN_LOG(trace, "Failed to read socket receive buffer size", *this);
+  }
+  return result;
+}
+
+Api::SysCallIntResult DynamicSocketOptionsImpl::setSocketRecvLoWat(int low_watermark) const {
+  int rc = -1;
+  int errsrv = -1;
+  socklen_t option_len = sizeof(int);
```

Comment:
`sizeof(low_watermark)` to make it a bit safer

Reply:
I will fix this.